File: app/src/test/java/com/example/proyectonieve/ui/PrecioProductoTest.java
package com.example.proyectonieve.ui;

import com.example.proyectonieve.data.Producto;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class PrecioProductoTest {

    @Test
    void precioPositivo() {
        Producto productoValido = new Producto(
                1L,
                10000,
                "Torta de Piña",
                "Torta de bizcocho de vainilla con trozos de piña",
                null,
                "Torta"
        );

        assertTrue(productoValido.getPrecio() > 0, "El precio del producto debe ser positivo");
    }

    @Test
    void precioNegativo() {
        Producto productoNoValido = new Producto(
                2L,
                -15000,
                "Torta de chocolate",
                "Torta de bizcocho de chocolate",
                null,
                "Torta"
        );

        assertTrue(productoNoValido.getPrecio() <= 0, "El precio es negativo o cero");
    }
}

File: app/src/test/java/com/example/proyectonieve/ui/PrecioProductoTest.kt
package com.example.proyectonieve.ui

import com.example.proyectonieve.data.Producto
import org.junit.Assert.assertTrue
import org.junit.Test

class PrecioProductoTest {

    @Test
    fun precioPositivo(){
        val productoValido = Producto(
            1,
            10000,
            "Torta de Piña",
            "Torta de bizcocho de vainilla con trozos de piña",
            null,
            "Torta"
        )
        val precio = productoValido.precio
        assertTrue("El precio del producto debe ser positivo",precio>0)
    }

    @Test
    fun precioNegativo(){
        val productoNoValido = Producto(
            1,
            -15000,
            "Torta de chocolate",
            "Torta de bizcocho de chocolate",
            null,
            "Torta"
        )
        val precioNegativo = productoNoValido.precio
        assertTrue("El precio es negativo o cero", precioNegativo<=0)
    }
}